pkg/transaction/eventgenerator.go

```diff
@@ -191,6 +200,123 @@ func (e *Generator) CreateEvents(summaryEvent LogEvent, detailEvents []LogEvent,
 	return events, nil
 }
 
+// getConsumerDetails - get the consumer information to add to transaction event.  If we don't have any
+// 		information we need to get the consumer information, then we just return nil
+func (e *Generator) getConsumerDetails(summaryEvent LogEvent) *ConsumerDetails {
+	cacheManager := agent.GetCacheManager()
+	appName := unknown
+	apiID := ""
+	stage := ""
+
+	if summaryEvent.TransactionSummary.Application != nil {
+		appName = summaryEvent.TransactionSummary.Application.Name
+		e.logger.
+			WithField("appName", appName).
+			Debug("transaction summary application name")
+	}
+
+	// get proxy information
+	if summaryEvent.TransactionSummary.Proxy != nil {
+		apiID = summaryEvent.TransactionSummary.Proxy.ID
+		stage = summaryEvent.TransactionSummary.Proxy.Stage
+		e.logger.
+			WithField("apiID", apiID).
+			WithField("stage", stage).
+			Debug("transaction summary proxy information")
+	} else {
+		return nil
+	}
+
+	// get the managed application
+	managedApp := cacheManager.GetManagedApplicationByName(appName)
+	if managedApp == nil {
+		e.logger.
+			WithField("appName", appName).
+			Debug("could not get managed application by name")
+		return nil
+	}
+	e.logger.
+		WithField("appName", appName).
+		WithField("managed app name", managedApp.Name).
+		Debug("managed application info")
```

Comment:
Thoughts on making the logger a local var and then continually adding the fields, so subsequent logs has all previous logs? Maybe overkill?

Also, should some of these be trace?

Reply:
I'll look at it... and yes

Reply:
I left the logging the way it is... maybe overkill?  I can be swayed if the other 2 have thoughts on it.

Reply:
...but i did update trace v debug